tests/conftest.py

```diff
@@ -248,3 +248,20 @@ def _inner(**kwargs):
         )
 
     return _inner
+
+
+@pytest.fixture(autouse=True)
+def reset_features():
```

Comment:
there is no need for this to be an independent fixture, is there?

Could this not be cleaner

```python
@pytest.fixture()
def enable_overlay_feature(reset_features):
    """Enable the overlay feature."""
    from craft_parts import Features

    Features.reset()
    Features(enable_overlay=True)
    yield
    Features.reset()
```

?

Reply:
> there is no need for this to be an independent fixture, is there?

There is, actually. If this feature is not used in *all* tests, then two consecutive tests that call `Application.run()` will try to enable the overlay feature twice and fail. It's a side-effect of the `Features` singleton.
I add it as a parameter explicitly in the `enable_overlay_feature()` fixture to ensure correct order: first `reset_features` is setup, then `enable_overlay_feature`, then the test runs, then teardown in the reverse order